### ISSUE TITLE
How to implement weight decay towards the pre-trained model?

### ISSUE THREAD

**Issue by sedol1339** [huggingface/transformers#33909]
Hello, let me one question.

If using HF Trainer for supervised fune-tuning, how do I implement penalizing the distance between starting and current weights? This was shown to be effective in https://arxiv.org/abs/1909.11299

**Comment by niqodea** [comment]
There is no out-of-the-box option to specify weight decay against pretrained weights that I am aware of, but you could achieve it easily by adding the loss yourself.

For example, [here](https://huggingface.co/docs/transformers/en/training#training-loop) is an example torch training loop. You could patch it to include the differences between the initial parameters and current ones, square it, and sum to the total loss with a weight term.

**Comment by sedol1339** [comment]
> There is no out-of-the-box option to specify weight decay against pretrained weights that I am aware of, but you could achieve it easily by adding the loss yourself.
> 
> For example, [here](https://huggingface.co/docs/transformers/en/training#training-loop) is an example torch training loop. You could patch it to include the differences between the initial parameters and current ones, square it, and sum to the total loss with a weight term.

However, this is not transformers.Trainer, but instead a custom loop. So, with Trainer this is not achievable?

**Comment by ArthurZucker** [comment]
This seems like a custom usage, I don't think we support this out of the box!

**Comment by yuanx749** [comment]
I think you can subclass and override the `compute_loss` method of `Trainer`, like [this example](https://huggingface.co/docs/transformers/v4.45.2/en/trainer#customize-the-trainer).

**Comment by ArthurZucker** [comment]
We also updated the API to support updating the `self.loss_function` of each models!
